apps/nextjs-website/src/lib/footer.ts

```diff
@@ -1,28 +1,31 @@
 import { FooterProps as FooterPropsEC } from '@pagopa/pagopa-editorial-components/dist/components/Footer';
-import {
-  FooterLinksType,
-  LinkType,
-} from '@pagopa/pagopa-editorial-components/dist/components/Footer/types';
+import * as icons from '@mui/icons-material';
 import { Footer } from './fetch/footer';
+
+type IconNames = keyof typeof icons;
 export type FooterProps = Omit<FooterPropsEC, 'onLanguageChanged'>;
 
-const generateLinks = (
-  section: Footer['data']['attributes']['links_aboutUs'],
-  isSocialLink = false
-): ReadonlyArray<FooterLinksType> =>
-  section.links
-    .filter((link) =>
-      isSocialLink
-        ? link.linkType === 'social' && link.icon
-        : ['internal', 'external'].includes(link.linkType)
-    )
-    .map((link) => ({
-      label: link.text ?? link.href,
-      href: link.href,
-      ariaLabel: link.ariaLabel ?? link.href,
-      linkType: isSocialLink ? 'external' : (link.linkType as LinkType),
-      ...(isSocialLink && { icon: link.icon }),
-    }));
+type Link = Footer['data']['attributes']['links_aboutUs']['links'][0];
+
+const makeLink = ({ text, linkType, ariaLabel, href }: Link) => {
+  const base = {
+    label: text ?? href,
+    href,
+    ariaLabel: ariaLabel ?? href,
+  };
+  return linkType !== 'social' ? [{ ...base, linkType }] : [];
+};
+
+const makeSocialLink = ({ text, icon, linkType, ariaLabel, href }: Link) => {
+  const base = {
+    label: text ?? href,
+    href,
+    ariaLabel: ariaLabel ?? href,
+    // This cast is required until the task B2BP-271 is done
+    ...(icon && { icon: icon as IconNames }),
```

Comment:
Icon should be mandatory for socialLinks

Reply:
It _is_ optional by code.
But leaving `icon` out means the entire socialLink won't be rendered.

Honestly the result is the same whether we don't return a socialLink that has no icon or if we return the socialLink even though it has no icon, so if we want to keep it the way it is right now it's fine by me.